src/Evolution/Systems/CurvedScalarWave/BoundaryConditions/BaylissTurkel.hpp

```diff
@@ -0,0 +1,161 @@
+template <size_t Dim, typename Frame>
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace CurvedScalarWave::BoundaryConditions {
+
+/*!
+ * \brief Implements constraint-preserving boundary conditions with a second
+ * order Bayliss-Turkel radiation boundary condition.
+ *
+ * \details The Bayliss-Turkel boundary conditions are technically only valid in
+ * flat space and should therefore only be used at boundaries where the
+ * background spacetime is approximately Minkwoski such as (sufficiently far
+ * out) outer boundaries for asymptotically flat spacetimes. Small reflections
+ * are still likely to occur.
+ *
+ * The constraint-preserving part of the boundary conditions are set on the time
+ * derivatives of the evolved fields according to Holst(2004). The physical
+ * Bayliss-Turkel boundary conditions are additionally set onto the time
+ * derivative of \f$\Pi\f$.
+ *
+ * The constraints are defined as follows:
+ *
+ * \f{align*}{
+ *  \mathcal{C}_i&=\partial_i\Psi - \Phi_i=0, \\
+ *  \mathcal{C}_{ij}&=\partial_{[i}\Phi_{j]}=0
+ * \f}
+ *
+ * The boundary conditions are then given by:
+ *
+ * \f{align*}{
+ * \partial_{t} \Psi&\to\partial_{t}\Psi +
+ *                    \lambda_\Psi n^i \mathcal{C}_i, \\
+ * \partial_{t}\Pi&\to\partial_{t}\Pi-\left(\partial_t\Pi
+ *                  - \partial_t\Pi^{\mathrm{BC}}\right)
+ *                  +\gamma_2\lambda_\Psi n^i \mathcal{C}_i
+ *                  =\partial_t\Pi^{\mathrm{BC}}
+ *                  +\gamma_2\lambda_\Psi n^i \mathcal{C}_i, \\
+ * \partial_{t}\Phi_i&\to\partial_{t}\Phi_i+
+ *                     \lambda_0n^jP^k{}_i\mathcal{C}_{jk}
+ *   = \partial_{t}\Phi_i+ \lambda_0n^j \mathcal{C}_{ji}.
+ * \f}
+ *
+ * The Bayliss-Turkel boundary conditions are given by
+ *
+ * \f{align*}{
+ *  \prod_{l=1}^m\left(\partial_t + \partial_r + \frac{2l-1}{r}\right)\Psi=0,
+ * \f}
+ *
+ * which we expand here to second order (\f$m=2\f$) to derive conditions for
+ * \f$\partial_t\Pi^{\mathrm{BC}}\f$:
+ *
+ * \f{align*}{
+ *  \partial_t\Pi^{\mathrm{BC}}
+ *   &=\left(\partial_t\partial_r + \partial_r\partial_t +
+ *     \partial_r^2+\frac{4}{r}\partial_t
+ *     +\frac{4}{r}\partial_r + \frac{2}{r^2}\right)\Psi \notag \\
+ *     &=\left((2n^i + \beta^i) \partial_t \Phi_i + n^i n^j\partial_i\Phi_j +
+ *     \frac{4}{r}\partial_t\Psi + \frac{4}{r}n^i\Phi_i + \frac{2}{r^2}\Psi
+ * \right) / \alpha.
+ * \f}
+ *
+ *
+ * This derivation makes the following assumptions:
+ *
+ * - The lapse, shift, normal vector and radius are time-independent,
+ * \f$\partial_t \alpha = \partial_t \beta^i = \partial_t n^i = \partial_t r =
+ * 0\f$. If necessary, these time derivatives can be accounted for in the future
+ * by inserting the appropriate terms in a straightforward manner.
+ *
+ * - The outer boundary is spherical. It might be possible to generalize this
+ * condition but we have not tried this.
+ */
+template <size_t Dim>
+class BaylissTurkel final : public BoundaryCondition<Dim> {
```

Comment:
[optional] maybe match this name with the flat space scalar wave case? I don't have a preference on which one is "better". This is certainly shorter, while the flat space name is maybe "more descriptive" :shrug:

Reply:
I changed it to be consistent